test/pkg/environment/common/expectations.go

```diff
@@ -373,6 +384,23 @@ func (env *Environment) EventuallyExpectPendingPodCount(selector labels.Selector
 	}).Should(Succeed())
 }
 
+func (env *Environment) EventuallyExpectBoundPodCount(selector labels.Selector, numPods int) []*v1.Pod {
+	GinkgoHelper()
+	var res []*v1.Pod
+	Eventually(func(g Gomega) {
+		res = []*v1.Pod{}
+		podList := &v1.PodList{}
+		g.Expect(env.Client.List(env.Context, podList)).To(Succeed())
```

Comment:
Can we input the selector into the list call so that we're not doing a cluster wide list?

Reply:
Good call. Don't know why I didn't do that from the start. That's a lot better here.